### DISCUSSION TITLE
AuthPTagHelper for html anchor tag

### DISCUSSION THREAD

**Discussion by hatuadevops** [JonPSmith/AuthPermissions.AspNetCore#50] [Ideas]
@JonPSmith  I thought of this whiles migrating an asp.net mvc 5 project to asp.net core 6. 

```c#[HtmlTargetElement(Attributes = "authp-haspermission")]
    public class AuthPTagHelper : TagHelper
    {
        private readonly IHttpContextAccessor httpContextAccessor;

        [HtmlAttributeName("authp-haspermission")]
        public Example3Permissions HasPermissions { get; set; }

        public AuthPTagHelper(IHttpContextAccessor httpContextAccessor)
        {
            this.httpContextAccessor = httpContextAccessor;
        }
        public override void Process(TagHelperContext context, TagHelperOutput output)
        {
            var hasPermission = httpContextAccessor.HttpContext.User.HasPermission(HasPermissions);
            if (!hasPermission)
            {
                output.SuppressOutput();
            }
        }
    }
```

Usage: 
`<a authp-haspermission="RoleChange" asp-controller="Roles" asp-action="Edit" asp-route-roleName="@item.RoleName">Edit</a>`

**Comment by JonPSmith** [comment]
Nice!

If you want to add this via a pull request, then put it in the [AuthPermissions.BaseCode.PermissionsCode](https://github.com/JonPSmith/AuthPermissions.AspNetCore/tree/main/AuthPermissions.BaseCode/PermissionsCode) namespace and add a test in the [TestPermissionChecker](https://github.com/JonPSmith/AuthPermissions.AspNetCore/blob/main/Test/UnitTests/TestAuthPermissions/TestPermissionChecker.cs).

Or you are not used to pull requests I can add this when I update the library.

**Reply by akudjawu** [reply]
@JonPSmith pardon me I noticed I use a different github account to start the thread. 
I would use this account for the pull request and possibly future contributions. 

Sure I can handle the pull request.

Few questions. 

With respect to placing AuthPTagHelper here [AuthPermissions.BaseCode.PermissionsCode](https://github.com/JonPSmith/AuthPermissions.AspNetCore/tree/main/AuthPermissions.BaseCode/PermissionsCode), there are two dependences

1. [microsoft.aspnetcore.http.abstractions](https://www.nuget.org/packages/Microsoft.AspNetCore.Http.Abstractions/)
2. [microsoft.aspnetcore.razor](https://www.nuget.org/packages/Microsoft.AspNetCore.Razor/) 

TagHelper is an abstract class in the namespace "Microsoft.AspNetCore.Razor.TagHelpers"
Are you okay adding these dependencies in that project.

**Reply by JonPSmith** [reply]
Sorry, my suggestion wasn't a good one. All the ASP.NET Core code is contained in the `AuthPermissions.AspNetCore` project which has the dependences you need. I suggest putting your code in the [AuthPermissions.AspNetCore.PolicyCode](https://github.com/JonPSmith/AuthPermissions.AspNetCore/tree/main/AuthPermissions.AspNetCore/PolicyCode) folder. 

NOTE: I'm not releasing a new version straight away so your new feature won't be out immediately.

**Reply by akudjawu** [reply]
No worries , no rush.
